best-practices.md

```diff
@@ -7,25 +7,30 @@ This guide assumes you are already familiar with the concepts and `Resource` API
 In order to encapsulate the functionality, the `Resource` interface is sometimes not implemented directly by the component but we rather create an (anonymous) inner class. However it is not sufficient to pass this resource to the `Context.register()` method; Global Context tracks resources using *weak* references. As there is no `unregister` method on the Context, had a strong reference been used this would prevent the component from being garbage-collected when the application releases it. Therefore the class implementing `Resource` should be stored inside the component (in a field) to prevent garbage-collection:
 
 ```java
-public class Component {
-    private final Resource cracHandler;
+import jdk.crac.*;
 
-    public Component() {
-        /* other initialization */
-        cracHandler = new Resource() {
+public class Component {
+    public static void main(String[] args) {
```

Comment:
After it was compiling it still was not working as for unknown reason nobody was calling constructor of `Component`.